resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshot.scala

```diff
@@ -86,10 +86,13 @@ object ExecutorPodsSnapshot extends Logging {
             sparkContainerStatusOpt match {
               case Some(sparkContainerStatus) =>
                 sparkContainerStatus.getState.getTerminated match {
+                  // If there is no terminated state we are running.
+                  case null => PodRunning(pod)
```

Comment:
Could you describe `NullPointerException` more explicitly, please?
```
java.lang.NullPointerException
	at org.apache.spark.scheduler.cluster.k8s.ExecutorPodsSnapshot$.toState(ExecutorPodsSnapshot.scala:89)
	at org.apache.spark.scheduler.cluster.k8s.ExecutorPodsSnapshot$.$anonfun$toStatesByExecutorId$1(ExecutorPodsSnapshot.scala:62)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:238)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike.map(TraversableLike.scala:238)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:231)
	at scala.collection.immutable.List.map(List.scala:298)
```

Reply:
Sure, so the NPE is triggered if there is no terminated state when we . Do we want to put that in the comments?

Reply:
The PR title is enough~